easybuild/tools/github.py

```diff
@@ -1591,6 +1591,24 @@ def update_pr(pr_id, paths, ecs, commit_msg=None):
     print_msg(msg, log=_log)
 
 
+def check_online_status():
+    """
+    Check whether we currently are online
+    Return True if online, else an URLError instance
+    """
+    # Try repeatedly and with different URLs to cater for flaky connections or servers
+    # Timeout and repeats set so total 1 minute
+    urls = [GITHUB_URL, GITHUB_API_URL]
+    for i in range(6):
```

Comment:
@Flamefire This may be a step too far...

If the connection is flaky, we should recognize it as such, and not try again with increasingly larger timeout settings imho.

The `406` issue we're seeing is probably just because we're opening `github.com` rather than `api.github.com`?

Reply:
> If the connection is flaky, we should recognize it as such, and not try again with increasingly larger timeout settings imho.

The timeout stays the same and isn't even triggered for the current issue. I could locally reproduce the the 406 is flaky: Running it multiple times works, so it might be a server issue with that specific URL.

> The 406 issue we're seeing is probably just because we're opening github.com rather than api.github.com

Why wouldn't github.com be correct? We use it in `download_repo`. But seems we mostly use GITHUB_API_URL so maybe we should check for that?

Reply:
Well, in theory both should be fine, shouldn't matter much...

We can even check `google.com` instead (but that makes less sense).

It's very unclear why we're even hitting the 406 all of a sudden, this has never popped up before, and the way we're checking whether we're online hasn't changed recently.

I'm reluctant to start using a creative workaround for something being broken upstream...

Reply:
Ok, then let's use the API-URL